options/m_option.c

```diff
@@ -2648,21 +2648,31 @@ const m_option_type_t m_option_type_channels = {
 
 static int parse_timestring(struct bstr str, double *time, char endchar)
 {
-    int a, b, len;
-    double d;
+    int h, m, len;
+    double s;
     *time = 0; /* ensure initialization for error cases */
-    if (bstr_sscanf(str, "%d:%d:%lf%n", &a, &b, &d, &len) >= 3)
-        *time = 3600 * a + 60 * b + d;
-    else if (bstr_sscanf(str, "%d:%lf%n", &a, &d, &len) >= 2)
-        *time = 60 * a + d;
-    else if (bstr_sscanf(str, "%lf%n", &d, &len) >= 1)
-        *time = d;
-    else
+    bool neg = bstr_eatstart0(&str, "-");
+    if (bstrchr(str, '-') >= 0)
+        return 0; /* string should not contains more '-' after this point */
```

Comment:
Might as well handle the `-+1:00` case.

```suggestion
    bool neg = bstr_eatstart0(&str, "-");
    if (!neg)
        bstr_eatstart0(&str, "+");
    if (bstrchr(str, '-') >= 0 || bstrchr(str, '+') >= 0)
        return 0; /* string should not contains more '-' or '+' after this point */
```

Other than this, seems fine.